controller/src/lib.rs

```diff
@@ -73,18 +78,76 @@ async fn spawn_backend(
     }
 }
 
-/// get backend associated with a lock or error
-fn backend_of_lock(
-    state: &StateHandle,
-    cluster_name: &ClusterName,
+async fn wait_for_locked_backend_assignment(
+    state: &mut StateHandle,
     lock: &str,
+    cluster_name: &ClusterName,
+    nats: &TypedNats,
 ) -> anyhow::Result<BackendId> {
-    state
+    let mut sub = nats
+        .subscribe_jetstream_subject(SubscribeSubject::<WorldStateMessage>::new(format!(
+            "state.cluster.{}.backend.*.assignment.lock.{}",
```

Comment:
Let's move the creation of this `SubscribeSubject` onto a method of `WorldStateMessage`, so that all details of nats subject mappings live in `core/src/messages/`. This makes it easier for us to figure out subject mappings and allows us to use the type system (instead of attempting to string match) to see where each subject is used.